### PR TITLE
screen: update to 4.6.2

### PR DESCRIPTION
**Pull request by dgilman**
Big reorganization of how this portfile gets patched. I went back to
Apple's screen-22 source, extracted their patches, rebased them against
latest, and put them together in a single file patch-apple.diff which
contains all of them. I think this will be more manageable going
forward.

I figured out the purpose of patch-config.h.in.diff and left a note in
that file explaining what it does.

I removed support for the MacOS X 10.5 patches. I have no machine to
test this and those are pretty old systems anyway.

Patches I removed:
* patch-Makefile.in.diff: Upstream builds just fine without this, so
removing them. It stripped CFLAGS from a build line, no idea what the
offending CFLAGS might have been.
* patch-configure-libelf.diff: This is a fix to a ./configure check for
the existence of SVR4. We are not SVR4, who cares if it is broken.
patch-configure.diff: More fixes to ./configure to detect utmp support.
Apple's patch uses /usr/bin/login to handle utmp so no need for
this.
* patch-marxargs.diff: Not used in old portfile.
* patch-process.c-fix-bogus-strncpy-size.diff: Fixed upstream.
* patch-pty.c.diff: Not used in old portfile.
* patch-screen.c-leopard.diff: Removed 10.5 support.
* patch-utmp.c: This appears to be a half-hearted attempt at fixing up
utmp support under MacOS X. Unfortunately for the patch author MacOS X
does not have a utmp, it uses utmpx, and we disable support for utmp
editing at compile time. This patch is pointless and was removed.
* patch-windowsize.diff: This patch has been updated but in a slightly
different form.

Patches that are now part of patch-apple.diff:
* patch-screen.c.diff
* patch-window.c.diff

Closes: https://trac.macports.org/ticket/44279
Closes: https://trac.macports.org/ticket/53355
Closes: https://trac.macports.org/ticket/51395

###### Type(s)
- [x] bugfix
- [x] enhancement

###### Tested on
macOS 10.13.6 17G65
Xcode 9.4.1 9F2000

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: sysutils/screen/Portfile [hunk 1]
```diff
@@ -29,18 +28,19 @@ master_sites        gnu:${name} \
 distfiles-append    18:encoding
 
 checksums           ${distname}${extract.suffix} \
-                    rmd160  9de183f86e36b211d56c35e58598728f332faec9 \
-                    sha256  fa4049f8aee283de62e283d427f2cfd35d6c369b40f7f45f947dbfd915699d63 \
+                    rmd160  371828d8de1f2f8a77b6c010e12b5ea91d351dd3 \
+                    sha256  1b6922520e6a0ce5e28768d620b0f640a6631397f95ccb043b70b91bb503fa3a \
+                    size    845210 \
                     18 \
```
Comment (pmetzger) [inline sysutils/screen/Portfile]: Pardon, what's "18", a file name?

Comment (dgilman) [reply sysutils/screen/Portfile]: Yes

**Comment by dgilman** [comment]
Not sure if i can legally upstream the apple patch. It is GPL2, but it is not GPL2+ so i imagine it can't be distributed with the gpl3 upstream. I am also not the copyright holder (apple is) so if the FSF has a contributor agreement i couldn't legally sign it.

**Comment by pmetzger** [comment]
Screen is managed by the FSF?

**Comment by kencu** [comment]
installs and works on 10.6.8 (built with clang-4.0 and configured to libc++)

**Pull request merged**